lib/shopify_api/resources/collection.rb

```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+module ShopifyAPI
+  class Collection < Base
+    include Events
+    include Metafields
+
+    def products(options = {})
+      Product.find(:all, from: "#{self.class.prefix}collections/#{id}/products.json", params: options)
```

Comment:
This should probably be behind it's own version check. There is no need to make a request if we can already tell that they are using the incorrect API version.

Reply:
👍